bin/setup-bench.mjs

```diff
@@ -0,0 +1,155 @@
+import 'zx/globals';
+import os from 'node:os';
+import { join } from 'node:path';
+
+/*
+
+  To run proper bench setup we need to do following things:
+
+  1.) Compile control packages
+  2.) Compile experiment packages
+  3.) Use SAME benchmark source
+      * we should be able to tweak bench
+        (add more cases, and still be able to compare with control)
+      * we should be able to re-run bench in CI from current branch with updated perf source
+
+*/
+
+const experimentBranchName =
+  process.env['EXPERIMENT_BRANCH_NAME'] || (await $`git rev-parse --abbrev-ref HEAD`).stdout.trim();
+const controlBranchName = process.env['CONTROL_BRANCH_NAME'] || 'main';
+const markers =
+  process.env['MARKERS'] ||
+  'navigationStart,renderStart,renderEnd,glimmer-render-1000-rows-start,glimmer-render-1000-rows-finished';
+const fidelity = process.env['FIDELITY'] || '20';
+const throttleRate = process.env['THROTTLE'] || '4';
+const FORK_NAME = process.env['FORK_NAME'] || '';
+
+const tempDir = os.tmpdir();
+
+const CONTROL_DIR = join(tempDir, 'control');
+const EXPERIMENT_DIR = join(tempDir, 'experiment');
+
+const CONTROL_BENCH_DIR = join(CONTROL_DIR, 'benchmark', 'benchmarks', 'krausest');
+const EXPERIMENT_BENCH_DIR = join(EXPERIMENT_DIR, 'benchmark', 'benchmarks', 'krausest');
+
+const pwdRaw = await $`pwd`;
+const pwd = pwdRaw.toString().trim();
+
+// we use benchmark from current commit, very useful if we need to tweak it
+const benchmarkFolder = 'benchmark';
+
+// remove node_modules from benchmark folder, maybe we could figure out better option to distribute bench source
+await $`rm -rf ${join(pwd, benchmarkFolder, 'node_modules')}`;
+await $`rm -rf ${join(pwd, benchmarkFolder, 'benchmarks', 'krausest', 'node_modules')}`;
+
+await $`rm -rf ${CONTROL_DIR}`;
+await $`rm -rf ${EXPERIMENT_DIR}`;
+await $`mkdir ${CONTROL_DIR}`;
+await $`mkdir ${EXPERIMENT_DIR}`;
+
+const isMacOs = os.platform() === 'darwin';
+
+const BENCHMARK_FOLDER = join(pwd, benchmarkFolder);
+
+const rawUpstreamUrl = await $`git ls-remote --get-url upstream`;
```

Comment:
where is upstream set?

Reply:
@NullVoxPopuli it's for developers who work with fork of `glimmerjs/glimmer-vm` (to be able to bench my personal branch locally against `main` in origin)

Reply:
but I mean, how does CI use it? I didn't see a `git remote add upstream ...` anywhere? maybe it's not needed?
I see this if block: https://github.com/glimmerjs/glimmer-vm/pull/1532/files/b23d20697db4bb8d68a69373a2cc41abd7512343#diff-1dbea00b97d62abb4c27f03038299813b6c09433bb293cfe31040c8714cdf4d0R60-R63

but I don't quite understand why you'd check if `upstreamUrl` === `upstream` 
to try myself, I tried on a repo that has no `upstream` remote, and saw this output:
```
❯ git ls-remote --get-url origin
git@github.com:NullVoxPopuli/limber.git

limber-3 on  main [$] 
❯ git ls-remote --get-url upstream
upstream
```

I understand now... but... wtf, git? lol

anywho, thanks! looks good